src/web-auth/web-message-handler.js

```diff
@@ -58,22 +59,22 @@ WebMessageHandler.prototype.run = function(options, cb) {
   ) {
     var error = err;
     if (!err && eventData.event.data.response.error) {
-      error = objectHelper.pick(eventData.event.data.response, ['error', 'error_description']);
+      error = eventData.event.data.response;
+    }
+    if (!error) {
+      var parsedHash = eventData.event.data.response;
+      return _this.webAuth.validateAuthenticationResponse(options, parsedHash, cb);
     }
     if (
-      error &&
       error.error === 'consent_required' &&
       windowHelper.getWindow().location.hostname === 'localhost'
     ) {
       _this.warn.warning(
         "Consent Required. Consent can't be skipped on localhost. Read more here: https://auth0.com/docs/api-auth/user-consent#skipping-consent-for-first-party-clients"
       );
     }
-    if (error) {
-      return cb(error);
-    }
-    var parsedHash = eventData.event.data.response;
-    _this.webAuth.validateAuthenticationResponse(options, parsedHash, cb);
+    _this.webAuth.transactionManager.clearTransaction(error.state);
```

Comment:
OK. This is the goal of this PR. However this `clearTransaction` won't get called if `error!=null` (see line 66, happy path). Is that fine?

Reply:
Yes, because then we'll call `validateAuthenticationResponse`, which needs the transaction to work (and it will be removed later)